frontend/src/app/general/breadcrumbs/breadcrumbs.service.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * SPDX-FileCopyrightText: Copyright DB Netz AG and the capella-collab-manager contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  Data,
+  NavigationEnd,
+  Router,
+} from '@angular/router';
+import { BehaviorSubject } from 'rxjs';
+import { filter } from 'rxjs/operators';
+
+export interface Breadcrumb {
+  label: string;
+  url: string;
+}
+
+@Injectable({
+  providedIn: 'root',
+})
+export class BreadcrumbsService {
+  private readonly _breadcrumbs = new BehaviorSubject<Breadcrumb[]>([]);
+
+  readonly breadcrumbs = this._breadcrumbs.asObservable();
+  private placeholders: Data = {};
+
+  constructor(private router: Router) {
+    this.router.events
+      .pipe(filter((event) => event instanceof NavigationEnd))
+      .subscribe(() => this.updateBreadcrumbs(router));
+  }
+
+  public updatePlaceholder(placeholders: Data) {
+    this.placeholders = {
+      ...this.placeholders,
+      ...placeholders,
+    };
+    this.updateBreadcrumbs(this.router);
+  }
+
+  private updateBreadcrumbs(router: Router) {
```

Comment:
```suggestion
  private updateBreadcrumbs() {
```

The router is always the same - Use `this.router` instead.